include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -476,9 +484,15 @@ class __future : private std::tuple<_Args...>
     auto
     get()
     {
-        wait();
         if constexpr (sizeof...(_Args) > 0)
-            return __get_value(std::get<0>(*this));
+        {
+            auto& __val = std::get<0>(*this);
+            if constexpr (decltype(__do_wait(__val))::value)
```

Comment:
Is it possible to use `std::is_same_v` instead of `__do_wait`?

Reply:
` std::is_same` requires type, but we have type template - sycl::buffer<T>